features/FEATURE_LWIP/lwip-interface/lwip-eth/arch/TARGET_Freescale/k64f_emac.c

```diff
@@ -526,15 +524,14 @@ static err_t k64f_low_level_output(struct netif *netif, struct pbuf *p)
 
   /* Wait until a descriptor is available for the transfer. */
   /* THIS WILL BLOCK UNTIL THERE ARE A DESCRIPTOR AVAILABLE */
-  while (g_handle.txBdCurrent->control & ENET_BUFFDESCRIPTOR_TX_READY_MASK)
-    osSemaphoreWait(k64f_enet->xTXDCountSem.id, osWaitForever);
+  osSemaphoreWait(k64f_enet->xTXDCountSem.id, osWaitForever);
```

Comment:
@kjbracey-arm, you make a good argument. I'm still hesitant, but suspect I'm weighing efficiency too heavily over robustness and consistency. 

I've pushed up a new patch that drops frames when the hardware buffers are exhausted. As expected, @infinnovation's example now only gets ~6 big packets through each iteration, but congestion control should be able to work around that.

I would be interested in seeing a performance comparison of the two strategies on a 10mbps link with something high level like tcp, but I don't currently have the hardware available.